### PR TITLE
[KYUUBI #3279][INFRA] Bump spotless-maven-plugin to 2.24.1

### PR DESCRIPTION
**Pull request by yikf**
### _Why are the changes needed?_

fix https://github.com/apache/incubator-kyuubi/issues/3279

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yikf** [comment]
cc @pan3793

File: pom.xml [hunk 1]
```diff
@@ -1970,6 +1970,9 @@
                     <artifactId>spotless-maven-plugin</artifactId>
                     <version>${maven.plugin.spotless.version}</version>
                     <configuration>
+                        <upToDateChecking>
+                            <enabled>true</enabled>
+                        </upToDateChecking>
```
Comment (pan3793) [inline pom.xml:1975]: what does "upToDate" mean?

Comment (yikf) [reply pom.xml:1975]: This is a new feature supported in version 2.18.0 and indicate incremental checking, see : https://github.com/diffplug/spotless/pull/935

**Review by pan3793 — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (pan3793) [reply pom.xml:1975]: Thanks

**Comment by pan3793** [comment]
Thanks, merging to master

**Pull request closed without merging**